packages/amplication-server/src/core/ai/btm-manager.service.ts

```diff
@@ -56,17 +53,25 @@ export class BtmManagerService {
                 !isDuplicatedAlreadyUsed
               );
             })
-            .map(
-              (dataModelName) =>
-                <BtmEntityRecommendation>{
-                  id: undefined,
-                  name: dataModelName,
-                  fields:
-                    originalResource.entities
-                      ?.find((x) => x.name === dataModelName)
-                      ?.versions[0]?.fields.map((field) => field.name) ?? [],
-                }
-            ),
+            .map((dataModelName) => {
+              const entityNameIdMap = originalResource.entities.reduce(
+                (map, entity) => {
+                  map[entity.name] = entity;
+                  return map;
+                },
+                {} as Record<string, EntityPartial>
+              );
+
+              return {
+                id: undefined,
```

Comment:
Is this the tempResourceId field? If it is, Why not generate a tempId like a random guid?